rust/connlib/clients/shared/src/lib.rs

```diff
@@ -27,6 +29,7 @@ use tokio::task::JoinHandle;
 /// A session is created using [Session::connect], then to stop a session we use [Session::disconnect].
 pub struct Session {
     channel: tokio::sync::mpsc::Sender<Command>,
+    dns_updater: tokio::sync::mpsc::UnboundedSender<Vec<IpAddr>>,
```

Comment:
Why have two channels? Can we just make the command one unbounded?

Reply:
I just didn't want to make the channel for the other events unbounded

Reply:
Why not? It is not like the app will memory DDoS connlib with a million commands that we can't process.

Reply:
If it won't, then set the limit to a million and panic if the impossible happens?

Reply:
Also it could be a `watch` or Notify+ArcSwap or something, right? No question of memory leaks when out-of-date DNS server vecs don't even matter.

Reply:
Bounded channels normally preemptively allocate enough to handle the worst case scenario, I don't think it'd make sense to set it to a million.

Also I think it'll look a bit weird and could throw someone off when debugging

Reply:
> Also it could be a watch or Notify+ArcSwap or something, right? No question of memory leaks when out-of-date DNS server vecs don't even matter.

I think this is what we discussed about having dumb plumbing, it's more logic that we can move inside the tunnel.

We could go with `watch` without the logic to notify if updated but that will also look a bit weird, having the notify/debounce logic distributed between the session and the tunnel.

Reply:
We are also only shipping around pointers or ZST here so the memory footprint should be tiny.

Lets set the bound to a 100 and panic on error. It should never be full in normal operation and if it is, that is a bug.

It will not be disconnected unless the app calls commands after `on_disconnect` which is also a bug.

Reply:
> It will not be disconnected unless the app calls commands after on_disconnect which is also a bug.

This can also happen if the user disconnects manually at the same time `on_disconnect` is called, super unlikely but possible.